src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModel.cs

```diff
@@ -3,8 +3,10 @@
     using System.Windows.Input;
     using Framework.Rx;
     using InstanceAdd;
+    using Validar;
     using Validation;
 
+    [InjectValidation]
```

Comment:
What sorcery is this?

Reply:
It is the [Validar Fody plugin](https://github.com/Fody/Validar)

Reply:
Yes, this injects IDataErrorInfo into the class.